proxy/proxy.go

```diff
@@ -0,0 +1,71 @@
+package proxy
+
+import (
+	"net/http"
+	"sync"
+
+	"github.com/gorilla/websocket"
+	"github.com/taskcluster/webhooktunnel/util"
+	"github.com/taskcluster/webhooktunnel/wsmux"
+)
+
+type Config struct {
+	Upgrader websocket.Upgrader
+	Logger   util.Logger
+}
+
+type proxy struct {
```

Comment:
You should export the type if you return it... `Proxy` and give it a documentation comment... you return it from `NewProxy` so should probably be exported...